module/zfs/ddt.c

```diff
@@ -503,11 +503,14 @@ ddt_get_dedup_histogram(spa_t *spa, ddt_histogram_t *ddh)
 {
 	for (enum zio_checksum c = 0; c < ZIO_CHECKSUM_FUNCTIONS; c++) {
 		ddt_t *ddt = spa->spa_ddt[c];
-		for (enum ddt_type type = 0; type < DDT_TYPES; type++) {
-			for (enum ddt_class class = 0; class < DDT_CLASSES;
-			    class++) {
-				ddt_histogram_add(ddh,
-				    &ddt->ddt_histogram_cache[type][class]);
+		if (ddt) {
+			for (enum ddt_type type = 0; type < DDT_TYPES; type++) {
+				for (enum ddt_class class = 0;
+				    class < DDT_CLASSES; class++) {
+					ddt_histogram_add(ddh,
+					    &ddt->ddt_histogram_cache
+					    [type][class]);
```

Comment:
I absolutely loathe this reformatting, but I loathe it less than any alternate formats I could come up with.

Suggestions welcome.

Reply:
One suggestion, change the initial for loop to:

for (enum ddt_type type = 0; type < DDT_TYPES && ddt, type++) {

Reply:
I debated it a bit, but I think ultimately I do prefer that.

Done.